src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -143,76 +147,82 @@ case class WdlNamespaceWithWorkflow(importedAs: Option[String],
-    load(wdlSource, resource, localImportResolver, None)
+    load(wdlSource, resource, Seq(fileResolver), None)
   }
 
   def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver): WdlNamespace = {
-    load(wdlSource, resource, importResolver, None)
+    load(wdlSource, resource, Seq(importResolver), None)
   }
 
-  private def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver, importedAs: Option[String]): WdlNamespace = {
+  private def load(wdlSource: WdlSource, resource: String, importResolver: Seq[ImportResolver], importedAs: Option[String]): WdlNamespace = {
     WdlNamespace(AstTools.getAst(wdlSource, resource), wdlSource, importResolver, importedAs, root = true)
   }
 
-  def apply(ast: Ast, source: WdlSource, importResolver: ImportResolver, namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
+  def apply(ast: Ast, source: WdlSource, importResolvers: Seq[ImportResolver], namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
     val imports = for {
       importAst <- Option(ast).map(_.getAttribute("imports")).toSeq
       importStatement <- importAst.astListAsVector.map(Import(_))
     } yield importStatement
 
-    /**
-      * For import statements with an 'as' clause, this translates to a sub-namespace
-      */
-    val namespaces: Seq[WdlNamespace] = for {
-      imp <- imports if imp.namespaceName.isDefined
-      source = importResolver(imp.uri)
-      if source.length > 0
-    } yield WdlNamespace.load(source, imp.uri, importResolver, imp.namespaceName)
+
+    def tryResolve(str: String, remainingResolvers: Seq[ImportResolver], errors: List[Throwable]): Option[WdlSource] = {
+      remainingResolvers match {
+        case resolver :: tail =>
+          Try(resolver(str)) match {
+            case Success(s) => Option(s)
+            case Failure(f) => tryResolve(str, tail, errors :+ f)
+          }
+        case Nil =>
+          val collectedErrors = errors.map(_.getMessage)
+          throw new ValidationException(s"Failed to import workflow $str from ", NonEmptyList.fromListUnsafe(collectedErrors))
+      }
+    }
 
     /**
-      * For import statements without an 'as' clause, this is treated as a c-style #include
+      * Translates all import statements to sub-namespaces
       */
-    val nonNamespacedImports = for {
-      imp <- imports if imp.namespaceName.isEmpty
-      source = importResolver(imp.uri)
-      ast = AstTools.getAst(source, imp.uri)
-      scopeAst <- ast.getAttribute("body").astListAsVector.collect({ case a: Ast => a })
-    } yield ast -> source
-
-    val topLevelScopeAsts = for {
-      namespaceAst <- nonNamespacedImports.map(_._1) :+ ast
-      scopeAst <- namespaceAst.getAttribute("body").astListAsVector.collect({ case a: Ast => a })
-    } yield scopeAst
+    val namespaces: Seq[WdlNamespace] = for {
+      imp <- imports
+      source = tryResolve(imp.uri, importResolvers, List.empty)
```

Comment:
Since `tryResolve` will fail if you try to pass in 0 resolvers (because the `Nil` case requires at least one error), make the `importResolvers` parameter a `NonEmptyList`